src/jit/flowgraph.cpp

```diff
@@ -22738,6 +22713,306 @@ void Compiler::fgRemoveEmptyFinally()
 }
 
 //------------------------------------------------------------------------
+// fgRemoveEmptyTry: Optimize try/finallys where the try is empty
+//
+// Notes:
+//    In runtimes where thread abort is not possible, `try {} finally {S}`
+//    can be optimized to simply `S`. This method looks for such cases and
+//    removes the try-finally from the EH table, making suitable flow, block
+//    flag, statement, and reigon updates.
+
```

Comment:
It might be useful to point out here why thread abort is an important limitation, e.g.:
```
// This optimization is not legal in runtimes that support thread abort because those runtimes
// ensure that a finally is completely executed before continuing to process the thread abort.
// With this optimization, the finally loses special status and such complete execution is
// not guaranteed
```
Or something like that.